src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1251,6 +1264,14 @@ def download(cls, pretrained_model_name, **kwargs) -> Union[str, os.PathLike]:
                 model_info_call_error = e  # save error to reraise it if model is not cached locally
 
         if not local_files_only:
+            filenames = {sibling.rfilename for sibling in info.siblings}
+            model_filenames, variant_filenames = variant_compatible_siblings(filenames, variant=variant)
+            # The variant filenames can have the legacy sharding checkpoint format that we check and throw
+            # a warning if detected.
+            if len(variant_filenames) > 0 and _check_legacy_sharding_variant_format(filenames, variant):
```

Comment:
Sorry, which error is being referred to here?

Reply:
Oh got it. I think it's here and should suffice:
https://github.com/huggingface/diffusers/blob/f583dad9bd4742b56264f236400bea1f71e74029/src/diffusers/pipelines/pipeline_utils.py#L757

But LMK if you think otherwise.

Reply:
Resolving becasue of https://github.com/huggingface/diffusers/pull/9253#discussion_r1770803517.